src/bindings/pygaia/classification/classificationtask.py

```diff
@@ -188,8 +197,8 @@ def run(self, className, outfilename, param, dsname, gtname, evalconfig):
 
 
 if __name__ == '__main__':
-    className, outfilename, param, dsname, gtname, evalconfig = cPickle.load(sys.stdin)
```

Comment:
I wonder if there's a better way of doing this - instead of passing parameters as a pickle.
If we used the `multiprocessing` to start a separate process then we should be able to pass arguments like a normal function call instead of manually using `subprocess`: https://github.com/MTG/gaia/blob/2337a347c792fbf8a515ce9bf96b773808eadca2/src/bindings/pygaia/classification/classificationtaskmanager.py#L208

Is there a specific reason why we want to keep this method of running the processes? Perhaps it's because separating tasks using subprocesses were a good way of ensuring that we didn't have bad memory issues?

I've been using [`concurrent.futures`](https://docs.python.org/3/library/concurrent.futures.html) a lot and it's really nice to use. There's a python 2 backport for it. Perhaps we could see if this fits our requirements.

Reply:
Actually, the code calling classificationTask as a function is already there, inside an if/else statement that it hardcoded to call it as a new subprocess.
https://github.com/MTG/gaia/blob/2337a347c792fbf8a515ce9bf96b773808eadca2/src/bindings/pygaia/classification/classificationtaskmanager.py#L205
I am not sure about the reason behind but I made a comparison in a small classification task using concurrent.futures between both approaches and calling it directly as a function was a little bit faster. Also, I didn't notice any difference in the memory status after any of the experiments. 

| Trial | As a function (s) | With subprocess (s) |
| ------------- |:-------------:| -----:|
| 1 | 46,28 | 46,94 |
| 2 | 46,09 | 46,83 |
| 3 | 46,31 | 46,96 |
| 4 | 46,46 | 46,36 |
| Mean | 46,28 | 46,77 |

I will set the subprocess call as a function by default and move to concurrent.futures.